src/packet.ml

```diff
@@ -309,7 +303,11 @@ module Tls_crypt = struct
     let op = op_key (operation p) key in
     match p with
     | `Ack hdr -> fst (to_be_signed_header op hdr)
+    | `Control (Hard_reset_client_v3, (hdr, msg_id, _wkc)) ->
+        (* HARD_RESET_CLIENT_V3 is special: the wkc is not considered part of the packet *)
+        to_be_signed_control op (hdr, msg_id, Cstruct.empty)
     | `Control (_, c) -> to_be_signed_control op c
+    | `Data _ -> assert false
```

Comment:
I think this is fine, but would appreciate a comment why it is fine -- i.e. "data packets are never processed with tls-crypt"

Reply:
I added comments to explain why. It would be nice to not have this case, but I failed to make it type check. Probably it's a case of adding `(_ :> Packet.pkt)` in the right places.